jdaviz/configs/default/plugins/model_fitting/fitting_backend.py

```diff
@@ -0,0 +1,53 @@
+from asteval import Interpreter
+
+from specutils.spectra import Spectrum1D
+from specutils.fitting import fit_lines
+
+__all__ = ['fit_model_to_spectrum']
+
+
+def fit_model_to_spectrum(spectrum, component_list, expression):
+    """
+    Fits an astropy CompoundModel to an instance of Spectrum1D.
+
+    Parameters
+    ----------
+    spectrum : :class:`specutils.spectrum.Spectrum1D`
+        The spectrum to be fitted.
+    component_list : list
+        Spectral model subcomponents stored in a list.
```

Comment:
It might be useful to specify here that the elements of the list need to be initialized astropy.modeling.models objects.

Reply:
Yes. And speaking of that, the initialization code should be part of the backend too. I think the old specviz code uses a similar architecture, where the GUI is generic enough and doesn't know any of the particulars of each specific model type, such as parameter names and the like. There is a bunch of initializer classes there that takes care of translating specifics into a generic architecture that the GUI can understand. Perhaps we should file a ticket for that?

Reply:
I had started working on a couple initializers in my fork (see [here](https://github.com/rosteen/jdaviz/blob/model_fitting_ui/jdaviz/configs/default/plugins/model_fitting/model_fitting.py), it should be easy enough to move those from the UI file to the backend.